sssd_test_framework/utils/pam.py

```diff
@@ -0,0 +1,290 @@
+            self.cmd = self.cmd + f"set {self.path}/access[{count}]/origin " + i["origin"] + "\n"
+            count = +count
+
+
+class PAMFaillock(PAMUtils):
+    """
+    Management of PAM Faillock on the client host.
+
+    .. code-block:: python
+        :caption: Example usage
+
+        @pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+        def test_example(client: Client, provider: GenericProvider):
+            # Add user
+            provider.user("user-1").add()
+
+            # Setup faillock
+            faillock = client.pam.faillock()
+            faillock.config_set({"deny": "3", "unlock_time": "300"})
+            faillock.config_write()
+
+            client.sssd.common.pam(["with-faillock"])
+
+            # Start SSSD
+            client.sssd.start()
+
+            # Check the results
+            assert client.auth.ssh.password("user-1", "Secret123")
+
+            # Three failed login attempts
+            for i in range(3):
+                assert not client.auth.ssh.password("user-1", "bad_password")
+
+            assert not client.auth.ssh.password("user-1", "Secret123")
+
+            # Reset user lockout
+            client.pam.faillock("user-1").reset
+            assert client.auth.ssh.password("user-1", "Secret123")
+    """
+
+    def __init__(self, util: PAMUtils, file: str, host: MultihostHost, fs: LinuxFileSystem) -> None:
+        """
+        :param util: PAMUtils object
+        :type util: PAMUtils
+        :param file: Configuration file
+        :type file: str
+        :param host: MultihostHost object
+        :type host: MultihostHost
+        :param fs: LinuxFileSystem object
+        :type fs: LinuxFileSystem
+        :param file: Faillock configuration file
+        :type file: str, optional
+        """
+        super().__init__(host, fs)
+        self._changed: bool = False
+
+        self.util: PAMUtils = util
+        self.file: str = file
+        self.path: str = "/files" + self.file
+        self.args: str = f'--noautoload --transform "Simplevars.lns incl {self.file}"'
+        self.cmd: str = ""
+
+        self._backup = f"{self.file}.bak.wfw"
+
+    def setup_when_used(self):
+        super().setup_when_used()
+        self.logger.info(f"Creating backup at '{self._backup}' on {self.host.hostname})")
+        self.util.host.ssh.run(f"cp -rf {self.file} {self._backup}")
+
+    def teardown_when_used(self):
+        super().teardown_when_used()
+        self.logger.info(f"Restoring backup at {self._backup} on {self.host.hostname}")
+        self.util.host.ssh.run(f"mv {self._backup} {self.file}")
+
+    def config_write(self) -> None:
+        """
+        Write faillock configuration.
+        :return: Self
+        :rtype: PAMFaillock
+        """
+        self.util.logger.info(f"Saving Augeas changes to {self.file} on {self.util.host.hostname}")
+        self.util.host.ssh.run(f'echo -e "{self.cmd}save\n" | augtool --echo {self.args}')
```

Comment:
I don't know precisely how augeas cli works but it should be able to take commands from stdin. So instead of using `echo`, you can pass the input in input `run(..., input=...)`